src/packet-ffxiv.c

```diff
@@ -145,8 +150,15 @@ static int dissect_frame(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, vo
     next_tvb = tvb_new_subset_remaining(tvb, FRAME_HEADER_LEN);
   }
 
-  tcp_dissect_pdus(next_tvb, pinfo, frame_tree, TRUE, BLOCK_HEADER_LEN,
-    get_message_length, dissect_message, data);
+  offset = 0;
+  for(int i = 0; i < header.blocks; ++i) {
+    message_tvb = tvb_new_subset_remaining(next_tvb, offset);
+    length = dissect_message(message_tvb, pinfo, frame_tree, data);
+    if(length < 0) {
```

Comment:
This should probably be `length < BLOCK_HEADER_LEN` since even if there is no payload in the message, the header still exists. If we have an incomplete header, something is wrong.